src/interfaces/ecpg/test/pgtypeslib/meson.build

```diff
@@ -0,0 +1,29 @@
+pgc_files = [
+  'dt_test',
+  'dt_test2',
+  'num_test',
+  'num_test2',
+  'nan_test',
+]
+
+foreach pgc_file: pgc_files
+  exe_input = custom_target(pgc_file,
+    input: '@0@.pgc'.format(pgc_file),
+    output: '@0@.c'.format(pgc_file),
+    command: [
+      ecpg_exe, 
+      '--regression',
+      '-I' + meson.current_source_dir(),
+      '-I' + meson.current_source_dir() + '/../../include/',
+      '-o', '@OUTPUT@',
+      '@INPUT@'
+    ],
+  )
+
+  executable(pgc_file,
+    exe_input,
+    dependencies: [frontend_code, libpq],
+    include_directories: [ecpg_inc],
+    link_with: [ecpglib, ecpg_compat, ecpg_pgtypes],
```

Comment:
It'd be good to mark this as "build_by_default: false". It might again be a good argument to replace most of the arguments with a kwargs: ecpg_test_exec_kw to avoid repeating all of this / needing to adjust multiple places when changing?

Reply:
These executables are needed for running ecpg tests. If I mark them as "build_by_default: false", tests fail.

Reply:
If they are needed by the tests, then they should be added as `depends: ...` to the `test()` definition.

Reply:
@eli-schwartz, we are generating .c files from .pgc files, then generating executables from these .c files. Executables and .c files are both needed by the tests. So, should `custom targets(for generating .c files from .pgc files)` be added as `depends` to the `executables()` definition?

Reply:
Adding the executable as a dependency of the test ensures that it is built before tuning the test (even if `build_by_default: false`).

The C files are already dependencies of the executable because they are in the sources, so they don't need to be added to the test also.

Reply:
Thanks! I have one more question, as far as I understand these executables will be created if they added as `depends:` to the `test()` right(No need to run tests)? And we are adding these executables as `depends:` to the `test()` as long as possible(We are trying to define all possible tests). So, for it to be useful should we have option like `-Dbuild_tests=true`?

Reply:
I'm not sure what you mean by this?

You're adding it to the depends to ensure that correct dependencies are set up, which is important for two reasons:
- you can `git bisect` and run a single test via `meson test test-name` which will not rebuild the entire project, only the dependencies for that specific test
- if you set `build_by_default: false` you need to add it as a test dep to make sure it is built before running the testsuite (let's ignore, for now, the fact that meson has a long-standing bug that test dependencies are re-added to the default target; this will eventually be fixed to only build it when running the test target, at which point an existing, well-defined project will immediately benefit)

Reply:
Thanks! Your answer has been explanatory.